app/controllers/posts_controller.rb

```diff
@@ -6,5 +6,25 @@ def index
   def show
     @author = User.find(params[:user_id])
     @post = @author.posts.find((params[:id]))
+    @comment = Comment.new
+    @like = Like.new
+  end
+
+  def new
+    @post = Post.new
+  end
+
```

Comment:
- [ ] OPTIONAL
It's good rails practice to use _strong parameters_ for query parameters, which should be similar to:

```ruby
# controllers/posts_controller.rb

  class PostsController < ApplicationController

       ...
    private 
    def post_params
      params.require(:post).permit(:title, :text)
    end

      ...

  end
```

Kindly read further lesson on  strong parameters via [this link](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides)